### DISCUSSION TITLE
Use FilePicker and FileSave to save multiple files.

### DISCUSSION THREAD

**Discussion by CliffAgius** [CommunityToolkit/Maui#1046] [New Feature Discussions]
Hi all,

I love the new FilePicker and FileSave additions so the toolkit and wanted to discuss maybe joining them so that you can use the FilePicker to get a folder and then FileSave to save more than one file.

Project I am currently working on takes data an produces PowerPoint files using the Syncfusion tools, but it can on some days need to create 100 files, using the FileSave methods is brilliant but the poor user has to select the folder and hit save for each and every one of them.  For a mobile I get this is a very rare use case but for desktop this is normal business needs read in data and process it and save the results.

I haven't looked into if this is possible just yet as didn't want to commit effort if the team didn't want to add this in so though starting a discussion here is a good start.

Cliff.

**Comment by VladislavAntonyuk** [comment]
Hi, could you propose API how it may look like?
Fotr now you can use folder picker to select the target folder and then using System.IO to save file to that folder

**Reply by bijington** [reply]
Did we discuss in the last standup the idea of FileSaver silently doing the saving? This could cover Cliffs scenario I think

**Reply by CliffAgius** [reply]
Yes it's the silent save I was looking for rather than having a Save Popup for each and every file.

Will System.IO work on all platforms the way you suggest?

I'm thinking we have an new method called SaveMultiple() where we just pass the Filename and stream and as long as it's been used at least once it knows the folder to save in.

This is all just an idea at present I have not fully thought it through yet but happy to if you think the idea has merit.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
